src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/MarkerVisual/MarkerVisualDetectorSpatialLocalizer.cs

```diff
@@ -70,49 +71,59 @@ public LocalizationSession(MarkerVisualDetectorSpatialLocalizer localizer, Marke
             /// <inheritdoc />
             protected override void OnManagedDispose()
             {
-                coordinateService.Dispose();
+                base.OnManagedDispose();
                 discoveryCTS.Dispose();
+                coordinateService.Dispose();
             }
 
             /// <inheritdoc />
-            public async Task<ISpatialCoordinate> LocalizeAsync(CancellationToken cancellationToken)
+            public override async Task<ISpatialCoordinate> LocalizeAsync(CancellationToken cancellationToken)
             {
-                DebugLog($"Waiting for marker visual, CanBeCanceled:{cancellationToken.CanBeCanceled}, IsCancellationRequested:{cancellationToken.IsCancellationRequested}");
-                await Task.WhenAny(coordinateAssigned.Task, Task.Delay(-1, cancellationToken));
-                if (string.IsNullOrEmpty(coordinateId))
+                if (!defaultCTS.Token.CanBeCanceled)
```

Comment:
>defaultCTS.Token [](start = 21, length = 16)

Instead of exposing the CTS, I would recommend an auto property of type CancellationToken that you set with the CTS token as soon as you create it. This way, if the CTS is cancelled and disposed, the token can be still used in it's cancelled state instead of throwing objectdisposedexception